npm test hangs infinitly on windows

## Steps to reproduce

- check out loopback-next repo
- `npm install`
- `npm test` or `DEBUG=mocha:* npm test`

## Current Behavior

Many tests are executed - one test does not end. `npm test` never ends and runs infinitely. 
Tried with `cmd`, `powershell` and git `bash`.

> mocha:runner run suite app-generator (SLOW) +17ms

^^^^^^is the last logged console output.

## Expected Behavior

`npm test` should end successfully or with some error message.

## Additional information

Windows 10 Pro ver 1909
node v10.15.3
Docker version 19.03.4, build 9013bf5
@loopback/cli version: 1.28.1

when executing `DEBUG=mocha:* npm test` these are the last few log messages:
```bash
..  mocha:runner run suite TodoApplication +60ms
  mocha:runnable timeout 30000 +11s
  mocha:runnable timeout 30000 +2s
...  mocha:runnable timeout 30000 +25ms
..  mocha:runner run suite TodoApplication when dealing with a single persisted todo +3s
........  mocha:runner run suite GeoLookupService +71ms
.  mocha:runner run suite TodoController +18ms
  mocha:runner run suite TodoController createTodo +0ms
..  mocha:runner run suite TodoController findTodoById +37ms
.  mocha:runner run suite TodoController findTodos +14ms
...  mocha:runner run suite TodoController replaceTodo +36ms
.  mocha:runner run suite TodoController updateTodo +12ms
.  mocha:runner run suite TodoController deleteTodo +11ms
.  mocha:runner run suite app-generator (SLOW) +17ms
  mocha:runnable timeout 60000 +235ms
```

Alternative to Win 10 native(or full VM) is to set up [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10).  VSCode has an [extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) to connect to remote.

Just run test and got following result:

```node
   2960 passing (4m)
  8 pending
  1 failing

  1) HttpServer (integration)
       reports error when the server cannot be started:
     AssertionError: expected [Promise] to be rejected
      at Assertion.fail (packages/testlab/node_modules/should/as-function.js:275:17)
      at /mnt/i/dev/loopback-next/packages/testlab/node_modules/should/as-function.js:1747:16
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:94:5)
      at Context.<anonymous> (packages/http-server/src/__tests__/integration/http-server.integration.ts:204:5)
```

I could write a small guide on setting this up as Windows users might need some hand holding on setting up env in linux.

--- a/packages/cli/test/integration/generators/app.integration.js
+++ b/packages/cli/test/integration/generators/app.integration.js
@@ -263,7 +263,11 @@ describe('app-generator with tilde project path', () => {
   const cwd = process.cwd();
 
   // If the test runs outside $home directory
-  if (process.env.CI && !process.env.DEBUG && tildify(sandbox) === sandbox) {
+  const runsOutsideRoot =
+    process.env.CI && !process.env.DEBUG && tildify(sandbox) === sandbox
+      ? true
+      : false;
+  if (runsOutsideRoot) {
     sandbox = path.join(os.homedir(), '.lb4sandbox/tilde-path-app');
     pathWithTilde = '~/.lb4sandbox/tilde-path-app';
   }
@@ -295,7 +299,12 @@ describe('app-generator with tilde project path', () => {
     // eslint-disable-next-line no-invalid-this
     this.timeout(30 * 1000);
 
-    process.chdir(sandbox);
+    // Handle special case - Skipping... not inside the project root directory.
+    if (runsOutsideRoot) {
+      process.chdir(sandbox);
+    } else {
+      process.chdir(rootDir);
+    }
     build.clean(['node', 'run-clean', sandbox]);
     process.chdir(cwd);
   });

--- a/packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts
+++ b/packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts
@@ -196,7 +196,10 @@ describe('HttpCachingProxy', () => {
     expect(result2).to.equal('2');
   });
 
-  it('handles the case where backend service is not running', async () => {
+  it('handles the case where backend service is not running', async function() {
+    // This test takes a bit longer to finish on windows.
+    // eslint-disable-next-line no-invalid-this
+    this.timeout(3000);
     await givenRunningProxy({logError: false});
 
     await expect(

--- a/packages/http-server/src/__tests__/integration/http-server.integration.ts
+++ b/packages/http-server/src/__tests__/integration/http-server.integration.ts
@@ -44,7 +44,10 @@ describe('HttpServer (integration)', () => {
       .expect(200);
   });
 
-  it('stops server', async () => {
+  it('stops server', async function() {
+    // This test takes a bit longer to finish on windows.
+    // eslint-disable-next-line no-invalid-this
+    this.timeout(3000);
     const serverOptions = givenHttpServerConfig();
     server = new HttpServer(dummyRequestHandler, serverOptions);
     await server.start();
@@ -52,7 +55,10 @@ describe('HttpServer (integration)', () => {
     await expect(httpGetAsync(server.url)).to.be.rejectedWith(/ECONNREFUSED/);
   });
 
-  it('stops server with grace period and inflight request', async () => {
+  it('stops server with grace period and inflight request', async function() {
+    // This test takes a bit longer to finish on windows.
+    // eslint-disable-next-line no-invalid-this
+    this.timeout(3000);
     const serverOptions = givenHttpServerConfig() as HttpServerOptions;
     serverOptions.gracePeriodForClose = 1000;
     const {emitter, deferredRequestHandler} = createDeferredRequestHandler();
@@ -77,7 +83,10 @@ describe('HttpServer (integration)', () => {
     await expect(httpGetAsync(server.url)).to.be.rejectedWith(/ECONNREFUSED/);
   });
 
-  it('stops server with shorter grace period and inflight request', async () => {
+  it('stops server with shorter grace period and inflight request', async function() {
+    // This test takes a bit longer to finish on windows.
+    // eslint-disable-next-line no-invalid-this
+    this.timeout(3000);
     const serverOptions = givenHttpServerConfig() as HttpServerOptions;
     serverOptions.gracePeriodForClose = 10;
     const {deferredRequestHandler} = createDeferredRequestHandler();

